docs/architecture/traffic-flow.md

```diff
@@ -0,0 +1,139 @@
+# Traffic Flow
+
+This diagram shows how user requests travel through NGINX Gateway Fabric.
+
+## Simple Overview
+
+```mermaid
+%%{init: {'theme':'dark', 'themeVariables': {'fontSize': '16px', 'darkMode': true, 'primaryColor': '#4f46e5', 'primaryTextColor': '#e5e7eb', 'primaryBorderColor': '#6b7280', 'lineColor': '#9ca3af', 'secondaryColor': '#1f2937', 'tertiaryColor': '#374151', 'background': '#111827', 'mainBkg': '#1f2937', 'secondBkg': '#374151', 'tertiaryTextColor': '#d1d5db'}}}%%
+graph LR
+    %% Simple traffic flow
+    USER[👤 User]
+
+    subgraph "Kubernetes Cluster"
+        NGINX[🌐 NGINX]
+
+        subgraph "Your Apps"
+            SVC1[🔵 user-service]
+            SVC2[🔵 order-service]
+            POD1[Pod A]
+            POD2[Pod B]
+            POD3[Pod C]
+        end
+    end
+
+    %% Simple flow
+    USER --> NGINX
+    NGINX --> SVC1
+    NGINX --> SVC2
+    SVC1 --> POD1
+    SVC1 --> POD2
+    SVC2 --> POD3
+
+    %% Dark-friendly styling
+    style USER fill:#fbbf24,stroke:#f59e0b,stroke-width:2px,color:#1f2937
+    style NGINX fill:#8b5cf6,stroke:#7c3aed,stroke-width:2px,color:#ffffff
+    style SVC1 fill:#10b981,stroke:#059669,stroke-width:2px,color:#ffffff
+    style SVC2 fill:#10b981,stroke:#059669,stroke-width:2px,color:#ffffff
+```
```

Comment:
nginx technically routes directly to the Pods, not through the Service. We use the Service to get information about the Pods.